homeassistant/components/sonarr/sensor.py

```diff
@@ -155,58 +179,73 @@ async def async_update(self) -> None:
         key = self.entity_description.key
 
         if key == "diskspace":
-            await self.sonarr.update()
+            self.data[key] = await self.sonarr.async_get_diskspace()
         elif key == "commands":
-            self.data[key] = await self.sonarr.commands()
+            self.data[key] = await self.sonarr.async_get_commands()
         elif key == "queue":
-            self.data[key] = await self.sonarr.queue()
+            self.data[key] = await self.sonarr.async_get_queue(
+                include_series=True, include_episode=True
+            )
         elif key == "series":
-            self.data[key] = await self.sonarr.series()
+            self.data[key] = await self.sonarr.async_get_series()
         elif key == "upcoming":
             local = dt_util.start_of_local_day().replace(microsecond=0)
             start = dt_util.as_utc(local)
             end = start + timedelta(days=self.upcoming_days)
 
-            self.data[key] = await self.sonarr.calendar(
-                start=start.isoformat(), end=end.isoformat()
+            self.data[key] = await self.sonarr.async_get_calendar(
+                start_date=start,
+                end_date=end,
+                include_series=True,
             )
         elif key == "wanted":
-            self.data[key] = await self.sonarr.wanted(page_size=self.wanted_max_items)
+            self.data[key] = await self.sonarr.async_get_wanted(
+                page_size=self.wanted_max_items,
+                include_series=True,
+            )
 
     @property
     def extra_state_attributes(self) -> dict[str, str] | None:
         """Return the state attributes of the entity."""
         attrs = {}
         key = self.entity_description.key
 
-        if key == "diskspace":
-            for disk in self.sonarr.app.disks:
-                free = disk.free / 1024**3
-                total = disk.total / 1024**3
+        if key == "diskspace" and self.data.get(key) is not None:
+            for disk in self.data[key]:
+                free = disk.freeSpace / 1024**3
+                total = disk.totalSpace / 1024**3
                 usage = free / total * 100
 
                 attrs[
                     disk.path
                 ] = f"{free:.2f}/{total:.2f}{self.unit_of_measurement} ({usage:.2f}%)"
         elif key == "commands" and self.data.get(key) is not None:
             for command in self.data[key]:
-                attrs[command.name] = command.state
+                attrs[command.name] = command.status
```

Comment:
@tkdrob so CI is hitting: AttributeError: 'Command' object has no attribute 'status' any idea how that is possible?

Reply:
nvm sorry was old fixture I think